packages/astro/test/test-utils.js

```diff
@@ -170,17 +177,17 @@ export async function loadFixture(inlineConfig) {
 	let devServer;
 
 	return {
-		build: (opts = {}) => {
+		build: async (opts = {}) => {
 			process.env.NODE_ENV = 'production';
-			return build(settings, { logging, telemetry, ...opts });
+			return build(await getSettings(), { logging, telemetry, ...opts });
```

Comment:
Do we need to call `getSettings` every time we call a function? Can't this be moved before the return at line 179?

Reply:
I documented why I did this at https://github.com/withastro/astro/blob/2bf1e2a9b88c2d6e9af93fb73f5e22a444a268bc/packages/astro/test/test-utils.js#L131-L142

The build/sync/check/dev commands could mutate `settings` so we'd have to re-create the settings to ensure the test isn't polluted.

Reply:
I read the comment, but I didn't understand that it was the reason why it has to be called every time. I thought it was the description of **what** the function does.

Reply:
I updated the comment at 8769d3c. Hope it's clearer now!